pkg/cmd/generate-cgo/main.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"text/template"
+
+	"github.com/cockroachdb/errors"
+
+	"github.com/alessio/shellescape"
+)
+
+type configuration struct {
+	Os   string
+	Arch string
+}
+
+var (
+	workspace                  string
+	bazelBin                   string
+	archivedCdepConfigurations = []configuration{
+		{"linux", "amd64"},
+		{"linux", "arm64"},
+		{"darwin", "amd64"},
+		{"darwin", "arm64"},
+		{"windows", "amd64"},
+	}
+)
+
+func init() {
+	flag.StringVar(&workspace, "workspace", "", "workspace path")
+	flag.StringVar(&bazelBin, "bazel-bin", "", "path to bazel/bin")
```

Comment:
Nit: Consider that `generate-cgo` can just call into `bazel info bazel-bin` itself, since it is the one doing the builds anyway, instead of requiring it to be passed in as an argument.